SpatialGDK/Source/Private/Interop/SpatialReceiver.cpp

```diff
@@ -312,13 +312,17 @@ void USpatialReceiver::ReceiveActor(Worker_EntityId EntityId)
 			}
 		}
 
-		// Add to entity registry. 
-		EntityRegistry->AddToRegistry(EntityId, EntityActor);
-
 		// Set up actor channel.
 		USpatialPackageMapClient* SpatialPackageMap = Cast<USpatialPackageMapClient>(Connection->PackageMap);
 		USpatialActorChannel* Channel = Cast<USpatialActorChannel>(Connection->CreateChannel(CHTYPE_Actor, NetDriver->IsServer()));
-		check(Channel);
+		if (!Channel)
+		{
+			EntityActor->Destroy(true);
```

Comment:
Is this what Unreal does what it runs out of channels on receiving an Actor?

Reply:
No, AFAIK natively it wouldn't spawn the actor if there are not enough channels. Also it is very unlikely to happen in native as the client won't have more channels than the server.
In our case I considered creating a channel before spawning the actor, but that doesn't work nice for the case when we're checking out a PlayerController.
Running out of channels already means things are going wrong, so I'm not sure what is the most graceful way of handling this.

Reply:
Worth adding a Warning log here?